File: trunk/CoherenceLoader/java/src/com/seovic/coherence/loader/source/CoherenceCacheSource.java
package com.seovic.coherence.loader.source;


import com.seovic.util.extractors.OgnlExtractor;
import com.seovic.util.Extractor;

import com.tangosol.net.CacheFactory;
import com.tangosol.net.NamedCache;

import java.util.Iterator;


/**
 * @author ic  2009.06.09
 */
public class CoherenceCacheSource
        extends AbstractBaseSource
    {
    private NamedCache cache;

    public CoherenceCacheSource(String cacheName)
        {
        this.cache = CacheFactory.getCache(cacheName);
        }

    public Iterator iterator()
        {
        return cache.values().iterator();
        }

    protected Extractor createDefaultExtractor(String propertyName)
        {
        return new OgnlExtractor(propertyName);
        }
    }

File: trunk/main.net/src/Coherence.Commons/Loader/Source/CoherenceCacheSource.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Text;
using Seovic.Coherence.Core;
using Tangosol.Net;

namespace Seovic.Coherence.Loader.Source
{
    public class CoherenceCacheSource : AbstractBaseSource
    {
        #region Constructors

        public CoherenceCacheSource(string cacheName)
        {
            cache = CacheFactory.GetCache(cacheName);
        }

        #endregion

        #region IEnumerable implementation

        /// <summary>
        /// Returns an enumerator that iterates through a collection.
        /// </summary>
        /// <returns>
        /// An <see cref="T:System.Collections.IEnumerator"/> object that can be used to iterate through the collection.
        /// </returns>
        /// <filterpriority>2</filterpriority>
        public override IEnumerator GetEnumerator()
        {
            return cache.Values.GetEnumerator();
        }

        #endregion

        #region AbstractBaseSource implementation

        protected override IExtractor CreateDefaultExtractor(string propertyName)
        {
            return Defaults.CreateExtractor(propertyName);
        }

        #endregion

        #region Data members

        private INamedCache cache;

        #endregion
    }
}